### PR TITLE
Initial app submission

### PR DESCRIPTION
**Pull request by crestdatasystems**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: actions/dataminrpulse_get_alerts.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+import dataminrpulse_consts as consts
+from actions import BaseAction
+
+
+class GetAlertsAction(BaseAction):
+    """Class to handle get alerts action."""
+
+    def execute(self):
+        """Execute the  get alerts action."""
```
Comment (alexa-phantom) [inline actions/dataminrpulse_get_alerts.py]:
```suggestion
        """Execute the get alerts action."""
```

File: actions/dataminrpulse_get_alerts.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        list_id = self._param.get("list_id", None)
+        use_asset_configured_lists = self._param.get("use_asset_configured_lists", False)
+        query = self._param.get("query", None)
+
+        # Accepts comma-seperated list only
+        if list_id:
+            list_id = list_id.strip(',')
+            list_id = str(list_id).replace(" ", '')
```
Comment (alexa-phantom) [inline actions/dataminrpulse_get_alerts.py]:
```suggestion
            list_id = str(list_id).replace(' ', '')
```
Use quote consistency please

File: actions/dataminrpulse_on_poll.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            # Storing updated configured values in state file
+            self._connector.state[consts.DATAMINRPULSE_STATE_LIST_ID_VALUE] = list_id
+            self._connector.state[consts.DATAMINRPULSE_STATE_QUERY_VALUE] = query
+
+        num = self._connector.config.get("page_size_for_polling", 40)
+        ret_val, num = self._connector.util._validate_integer(self._action_result, num, "num", True)
+        if phantom.is_fail(ret_val):
+            return self._action_result.get_status()
+
+        alert_type = self._connector.config.get("alert_type", 'All')
```
Comment (alexa-phantom) [inline actions/dataminrpulse_on_poll.py]: Quote consistency. Use either " or '

File: dataminrpulse.json [hunk 4]
```diff
@@ -0,0 +1,1978 @@
...
+            "read_only": true,
+            "parameters": {
+                "list_id": {
+                    "description": "List ID",
+                    "data_type": "string",
+                    "primary": true,
+                    "contains": [
+                        "dataminrpulse list id"
+                    ],
+                    "order": 0
```
Comment (alexa-phantom) [inline dataminrpulse.json:229]: `order` = `1` is missing in this list of parameters

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: actions/dataminrpulse_get_alerts.py (hunk 1 above)
Comment (pdesai-crest) [reply actions/dataminrpulse_get_alerts.py]: Done

File: actions/dataminrpulse_get_alerts.py (hunk 2 above)
Comment (pdesai-crest) [reply actions/dataminrpulse_get_alerts.py]: Done

File: actions/dataminrpulse_on_poll.py (hunk 3 above)
Comment (pdesai-crest) [reply actions/dataminrpulse_on_poll.py]: Done

File: dataminrpulse.json (hunk 4 above)
Comment (pdesai-crest) [reply dataminrpulse.json:229]: Done

**Review by ricardo-phantom — APPROVED** [review]

**Review by alexa-phantom — APPROVED** [review]

**Pull request merged**